### PR TITLE
Compression changes to eo-learn

### PR DESCRIPTION
**Pull request by mlubej**
Changes:
- `compression_level` parameter removed
- compress by default for everything except bbox and timestamps
- test updates

### CODE REVIEW

**Comment by mlubej** [comment]
@zigaLuksic turns out that backwards compatibility already worked out of the box, since the featureIO objects retained the ability to handled loading of compressed/uncompressed objects

I added back the compression_level parameter as discussed, and added the EODeprecation warning on the save task and the save method. I removed the parameter from the docs, though.

Lastly, I added tests for checking the compression deprecation warning, and augmented a test for checking backwards compatibility loading and proper cleanup if different compressions are present (from before)

**Comment by zigaLuksic** [comment]
> @zigaLuksic turns out that backwards compatibility already worked out of the box, since the featureIO objects retained the ability to handled loading of compressed/uncompressed objects

Slight correction, the `FeatureIO` objects retained the GZip capabilities so that we have backwards compatibility ;)
After a while, when this is removed, we'll be able simplify FeatureIO classes

**Comment by mlubej** [comment]
Also, I wasn't sure if I should keep the compression capabilities in the `save_eopatch` function (from eolearn/core/eodata_io.py). This one should be internal, so people shouldn't be using it, and the tests might've looked a bit better, but dunno, someone still might use it.

**Comment by zigaLuksic** [comment]
> Also, I wasn't sure if I should keep the compression capabilities in the `save_eopatch` function (from eolearn/core/eodata_io.py). This one should be internal, so people shouldn't be using it, and the tests might've looked a bit better, but dunno, someone still might use it.

I see no reason to keep it, since it would control only part of the compression behaviour (for numpy and geodata) which we dont really want to be changed (except for tests)

File: tests/core/test_eodata_io.py [hunk 1]
```diff
@@ -408,65 +420,23 @@ def test_cleanup_different_compression(fs_loader, eopatch):
...
-            folder, filesystem=temp_fs, compress_level=9, overwrite_permission="OVERWRITE_FEATURES", use_zarr=True
-        )
-        save_noncompressed_task = SaveTask(
-            folder, filesystem=temp_fs, compress_level=0, overwrite_permission="OVERWRITE_FEATURES", use_zarr=True
-        )
-        bbox_path = fs.path.join(folder, patch_folder, "bbox.geojson")
-        compressed_bbox_path = bbox_path + ".gz"
-        mask_timeless_path = fs.path.join(folder, patch_folder, "mask_timeless", "mask.zarr")
-        wrong_compressed_mask_timeless_path = mask_timeless_path + ".gz"
+        # need to manually save uncompressed feature
```
Comment (zigaLuksic) [inline tests/core/test_eodata_io.py:427]: i'd also add this for bbox and timestamps

File: tests/core/test_eodata_io.py [hunk 2]
```diff
@@ -408,65 +420,23 @@ def test_cleanup_different_compression(fs_loader, eopatch):
...
+        ftype, fname = (FeatureType.MASK_TIMELESS, "mask")
+        ftype_path = fs.path.join(folder, patch_folder, ftype.value)
+        temp_fs.makedirs(ftype_path)
+        feature_io = FeatureIONumpy(os.path.join(ftype_path, fname + ".npy"), filesystem=temp_fs)
+        feature_io.save(eopatch[(ftype, fname)], temp_fs, os.path.join(ftype_path, fname), compress_level=0)
```
Comment (zigaLuksic) [inline tests/core/test_eodata_io.py:432]: save methods are class methods, no need to instantiate the object `feature_io` (they are attached to FeatureIO due to organizational reasons)

File: tests/core/test_eodata_io.py [hunk 3]
```diff
@@ -520,17 +490,16 @@ def test_lazy_loading_plus_overwrite_patch(fs_loader, folder_name, eopatch, use_
         (FeatureIOTimestamps, [datetime.datetime(2017, 1, 1, 10, 4, 7), datetime.datetime(2017, 1, 4, 10, 14, 5)]),
     ],
 )
-@pytest.mark.parametrize("compress_level", [0, 1])
```
Comment (zigaLuksic) [inline tests/core/test_eodata_io.py:523]: why are you removing this parametrization here?

File: tests/core/test_eodata_io.py (hunk 1 above)
Comment (mlubej) [reply tests/core/test_eodata_io.py]: added

File: tests/core/test_eodata_io.py (hunk 2 above)
Comment (mlubej) [reply tests/core/test_eodata_io.py]: thanks, updated

Comment (mlubej) [reply tests/core/test_eodata_io.py]: btw, should I add here something that checks if `EOPatch.load` works with uncompressed features for backwards compatibility? 

just something like 

```python
EOPatch.load(fs.path.join(folder, patch_folder), filesystem=temp_fs)
```

File: tests/core/test_eodata_io.py (hunk 3 above)
Comment (mlubej) [reply tests/core/test_eodata_io.py]: my bad, reverted

File: tests/core/test_eodata_io.py [hunk 4]
```diff
@@ -328,6 +327,19 @@ def test_overwrite_failure(fs_loader, use_zarr: bool):
             )
 
 
+@mock_s3
+@pytest.mark.parametrize("fs_loader", FS_LOADERS)
+@pytest.mark.parametrize("compress_level", [0, 1])
+def test_compression_deprecation(eopatch, fs_loader, compress_level: int | None):
+    folder = "foo-folder"
+
```
[the remark this answers is not in the document]
Comment (mlubej) [reply tests/core/test_eodata_io.py:335]: done

File: tests/core/test_eodata_io.py (hunk 2 above)
Comment (zigaLuksic) [reply tests/core/test_eodata_io.py]: yes please <3

Comment (mlubej) [reply tests/core/test_eodata_io.py]: done

**Review by zigaLuksic — APPROVED** [review]

**Pull request merged**